desc/magnetic_fields.py

```diff
@@ -1034,6 +1004,238 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
+                    zero_due_to_sym_inds = [1, 2]  # b=c=0 for odd l with sym
+                for which_coef in range(4):
+                    if which_coef == 0:
+                        a_s.append(1)
+                    elif which_coef == 1:
+                        b_s.append(1)
+                    elif which_coef == 2:
+                        c_s.append(1)
+                    elif which_coef == 3:
+                        d_s.append(1)
+                    if which_coef in zero_due_to_sym_inds:
+                        abcd_zero_due_to_sym_inds[which_coef].append(0)
+                    else:
+                        abcd_zero_due_to_sym_inds[which_coef].append(1)
+
+                ms.append(m)
+                ls.append(l)
+        for i in range(4):
+            abcd_zero_due_to_sym_inds[i] = jnp.asarray(abcd_zero_due_to_sym_inds[i])
+        assert (
+            round(jnp.sum(len(a_s) + len(b_s) + len(c_s) + len(d_s))) == num_modes - 1
+        )
+        params = {
+            "ms": ms,
+            "ls": ls,
+            "a_arr": a_s,
+            "b_arr": b_s,
+            "c_arr": c_s,
+            "d_arr": d_s,
+            "B0": 0.0,
+        }
+        n = (
+            round(num_modes - 1) / 4
+        )  # how many l-m mode pairs there are, also is len(a_s)
+        n = int(n)
+        domm_field = DommaschkPotentialField(0, 0, 0, 0, 0, 0, 1)
+
+        def get_B_dom(coords, X, ms, ls):
+            """Fxn wrapper to find jacobian of dommaschk B wrt coefs a,b,c,d."""
+            # zero out any terms that should be zero due to symmetry, which
+            # we cataloged earlier for each a_arr,b_arr,c_arr,d_arr
+            # that way the resulting modes after pinv don't contain them either
+            return domm_field.compute_magnetic_field(
+                coords,
+                params={
+                    "ms": jnp.asarray(ms),
+                    "ls": jnp.asarray(ls),
+                    "a_arr": jnp.asarray(X[1 : n + 1]) * abcd_zero_due_to_sym_inds[0],
+                    "b_arr": jnp.asarray(X[n + 1 : 2 * n + 1])
+                    * abcd_zero_due_to_sym_inds[1],
+                    "c_arr": jnp.asarray(X[2 * n + 1 : 3 * n + 1])
+                    * abcd_zero_due_to_sym_inds[2],
+                    "d_arr": jnp.asarray(X[3 * n + 1 : 4 * n + 1])
+                    * abcd_zero_due_to_sym_inds[3],
+                    "B0": X[0],
+                },
+            )
+
+        X = []
+        for key in ["B0", "a_arr", "b_arr", "c_arr", "d_arr"]:
+            obj = params[key]
+            if isinstance(obj, list):
+                X += obj
+            else:
+                X += [obj]
+        X = jnp.asarray(X)
+
+        jac = jit(jacfwd(get_B_dom, argnums=1))(coords, X, params["ms"], params["ls"])
+
+        A = jac.reshape((rhs.size, len(X)), order="F")
+
+        # now solve Ac=b for the coefficients c
+
+        Ainv = scipy.linalg.pinv(A)
+        c = jnp.matmul(Ainv, rhs)
```

Comment:
`jnp.linalg.lstsq` would be more stable here, and automatically gives residuals, condition numbers etc.